pkg/api/types.go

```diff
@@ -1410,6 +1412,55 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// RateLimits is a list of rate limits on disruptions (preemption and other evictions) across the corresponding
+	// collection of pods, e.g. no more than one disruption per hour across the pods of a particular Service.
+	// If there are multiple entries in the list, all must be satisfied for a disruption to be allowed.
+	RateLimits   []RateLimit `json:"rateLimits,omitempty"`
+
+	MinInstances MinInstances `json:"minInstances,omitempty"`
+}
+
+type MinInstances struct {
+	// The minimum number of pods that must be available simultaneously.
+	// Value can be an absolute number (ex: 5) or a percentage of total pods(ex: 10%).
+	// Absolute number is calculated from percentage by rounding up.
+	MinAvailable intstr.IntOrString `json:"minAvailable"`
+	// Minimum number of seconds for which a newly created pod must be ready for it to be considered available.
+	// Defaults to 0 (pod will be considered available as soon as it is ready).
+	MinReadySeconds int `json:"minReadySeconds"`
```

Comment:
I do not feel MinReadySeconds is very descriptive. This is under `MinInstances` struct, so intuitively users assume this field describes MinInstances, but this actually defines `Available` (what do we mean Available for the first field in the struct). So I feel this is confusing.